File: run_test.py
from playwright.sync_api import sync_playwright

def run_tests():
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        page = browser.new_page()

        # Attach console listeners to capture any errors from the game loop
        page.on("console", lambda msg: print(f"Browser Console [{msg.type}]: {msg.text}"))
        page.on("pageerror", lambda err: print(f"Browser Error: {err}"))

        # Load the local index.html file
        page.goto("file:///app/index.html", wait_until="networkidle")

        # Wait a bit for game to initialize
        page.wait_for_timeout(2000)

        # Test finding and adding a new creature to saveCollected directly
        page.evaluate("""
            const testCreature = { id: 'test_123', name: 'Testimon' };
            saveCollected(testCreature);
            if (!collectedCreatures.find(c => c.id === 'test_123')) {
                console.error("Test failed: test_123 not in array");
            } else {
                console.log("Test passed: test_123 in array");
            }
            if (!collectedCreaturesIds.has('test_123')) {
                console.error("Test failed: test_123 not in Set");
            } else {
                console.log("Test passed: test_123 in Set");
            }
        """)

        browser.close()

if __name__ == "__main__":
    run_tests()

File: run_test.js
const { chromium } = require('playwright');
(async () => {
    const browser = await chromium.launch({ headless: true });
    const page = await browser.newPage();

    // Add console listener
    page.on('console', msg => console.log('BROWSER:', msg.text()));

    await page.goto('file://' + __dirname + '/index.html');
    await page.waitForTimeout(1000);

    // Evaluate in browser
    const res = await page.evaluate(async () => {
        return new Promise(resolve => {
            // Wait for game start
            document.getElementById('startGameBtn').click();
            setTimeout(() => {
                let scene = window.game.scene.scenes[0];
                let group = scene.obstaclesGroup;

                let firstObs = group.getChildren()[0];
                console.log("Obs body properties:");
                console.log("x, y:", firstObs.body.x, firstObs.body.y);
                console.log("width, height:", firstObs.body.width, firstObs.body.height);
                console.log("center x, y:", firstObs.body.center.x, firstObs.body.center.y);

                resolve({
                    x: firstObs.body.x,
                    y: firstObs.body.y,
                    width: firstObs.body.width,
                    height: firstObs.body.height
                });
            }, 1000);
        });
    });
    console.log(res);
    await browser.close();
})();
